daemon/oci_linux.go

```diff
@@ -474,7 +474,7 @@ func setMounts(daemon *Daemon, s *specs.Spec, c *container.Container, mounts []c
 			// filter out mount overridden by a user supplied mount
 			continue
 		}
-		if mountDev && strings.HasPrefix(m.Destination, "/dev/") {
+		if mountDev && strings.HasPrefix(m.Destination, "/dev/") && m.Destination != "/dev/init" {
```

Comment:
nitty; wondering if it would be clearer to make this a separate `if` (also if we should put `/dev/init` in a constant)